crates/sui-framework/packages/sui-framework/tests/test_scenario_tests.move

```diff
@@ -407,6 +407,258 @@ module sui::test_scenario_tests {
+        let id1_addr = object::uid_to_address(&uid1);
+        {
+            let obj1 = Object { id: uid1, value: 10 };
+            let obj2 = Object { id: uid2, value: 20 };
+            transfer::public_transfer(obj1, sender);
+            transfer::public_transfer(obj2, id1_addr);
+        };
+        ts::next_tx(&mut scenario, sender);
+        {
+
+            let parent = ts::take_from_sender_by_id<Object>(&scenario, id1);
+            let t2 = ts::receiving_ticket_for_most_recent_child_object<Object>(&id1);
+            let obj2 = transfer::receive(&mut parent.id, t2);
+            assert!(parent.value == 10, EValueMismatch);
+            assert!(obj2.value == 20, EValueMismatch);
+            ts::return_to_sender(&scenario, parent);
+            transfer::public_transfer(obj2, id1_addr);
+        };
+        ts::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = ts::EObjectNotFound)]
+    fun test_receive_object_shared() {
+        let sender = @0x0;
+        let scenario = ts::begin(sender);
+        let uid1 = ts::new_object(&mut scenario);
+        let id1 = object::uid_to_inner(&uid1);
+        {
+            let obj1 = Object { id: uid1, value: 20 };
+            transfer::public_share_object(obj1);
+        };
+        ts::next_tx(&mut scenario, sender);
+        {
+            let _t2 = ts::receiving_ticket_for_child_object_by_id<Object>(id1);
+        };
+        ts::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = ts::EReceivingTicketAlreadyAllocated)]
+    fun test_receive_object_double_allocate_ticket() {
+        let sender = @0x0;
+        let scenario = ts::begin(sender);
+        let uid1 = ts::new_object(&mut scenario);
+        let id1 = object::uid_to_inner(&uid1);
+        {
+            let obj1 = Object { id: uid1, value: 20 };
+            transfer::public_transfer(obj1, sender);
+        };
+        ts::next_tx(&mut scenario, sender);
+        {
+            let _t2 = ts::receiving_ticket_for_child_object_by_id<Object>(id1);
+            let _t2 = ts::receiving_ticket_for_child_object_by_id<Object>(id1);
+        };
+        ts::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = ts::EReceivingTicketAlreadyAllocated)]
+    fun test_receive_deallocate_non_ticket() {
+        let sender = @0x0;
+        let scenario = ts::begin(sender);
+        let uid1 = ts::new_object(&mut scenario);
+        let id1 = object::uid_to_inner(&uid1);
+        {
+            let obj1 = Object { id: uid1, value: 20 };
+            transfer::public_transfer(obj1, sender);
+        };
+        ts::next_tx(&mut scenario, sender);
+        {
+            let _t2 = ts::receiving_ticket_for_child_object_by_id<Object>(id1);
+            let _t2 = ts::receiving_ticket_for_child_object_by_id<Object>(id1);
+        };
+        ts::end(scenario);
+    }
+
+    #[test]
+    fun test_receive_double_allocate_ticket_return_between() {
```

Comment:
These tests could maybe benefit from a short comment explaining what they are testing -- I see there's an effort made to have the `name` do that job, but it's stretching the limits of what is feasible with a Move identifier today...

Reply:
It's like a tweet, how much can you fit into 128 characters...